docs/apache-airflow/concepts/timetable.rst

```diff
@@ -51,6 +51,12 @@ As such, Airflow allows for custom timetables to be written in plugins and used
 DAGs. An example demonstrating a custom timetable can be found in the
 :doc:`/howto/timetable` how-to guide.
 
+.. note::
+
+    As a general rule, always access Variables, Connections etc or anything that would access
+    the database as late as possible in your code. See :ref:`best_practices/top_level_code`
+    for more best practices to follow.
```

Comment:
We should add an example here for timetable specifically here. For users not very familiar with Python, there’s nothing in the referenced section that would jump out and indicate what’s wrong.

(Alternatively, we can add an example in the referenced section explaining that function argument defaults and class-level variables are also top level code.)